### PR TITLE
Don't use dashed Bash command function names.

### PR DESCRIPTION
**Pull request by mikemcquaid**
### Changes to Homebrew's Core:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031) if you'd like one.
- [x] Have you successfully ran `brew tests` with your changes locally?

These are not valid POSIX and seem to cause issues in some cases e.g. issue #49514.

CC @xu-cheng @bfontaine @UniqMartin

### CODE REVIEW

File: Library/brew.sh [hunk 1]
```diff
@@ -167,7 +167,7 @@ then
   # Hide shellcheck complaint:
   # shellcheck source=/dev/null
   source "$HOMEBREW_BASH_COMMAND"
-  { "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }
+  { "homebrew_$HOMEBREW_COMMAND" "$@"; exit $?; }
```
Comment (xu-cheng) [inline Library/brew.sh]: Should we also escape `HOMEBREW_COMMAND`? There is no need for now but may help in the future.

Comment (mikemcquaid) [inline Library/brew.sh]: like `${HOMEBREW_COMMAND}`?

Comment (xu-cheng) [inline Library/brew.sh]: No, I mean escape dash inside HOMEBREW_COMMAND to underline. i.e.

`${HOMEBREW_COMMAND//-/_}`

**Comment by mikemcquaid** [comment]
Also added a commit to unset POSIX mode if set by e.g. `POSIXLY_CORRECT=1`.

**Comment by mikemcquaid** [comment]
Decided to back out on the underline changes and only make the required fix: 8313809.

**Comment by UniqMartin** [comment]
:+1: Sorry for the late response and thanks for resolving it so quickly! I'm still amazed in how many ways a user's environment can mess with code that works flawlessly in normal circumstances …

**Comment by mikemcquaid** [comment]
@UniqMartin Yeh, it's crazy, eh? I'd expect this environment variable to break almost every Bash script using bashisms (which is most of them).

**Pull request closed without merging**